### ISSUE TITLE
PersistedModel's updateAll method crashes the server when invoked via the remote connector

### ISSUE THREAD

**Issue by angfal** [strongloop/loopback#3717]
# Description

I have a model with a remote datasource. When I tries to execute `updateAll` method my server crashes. 

The problem is loopback registers `updateAll` remote method with an object type, which is not supported by strong-remoting type registry.

```
setRemoting(PersistedModel, 'updateAll', {
   ...
   returns: {
     arg: 'info',
     description: 'Information related to the outcome of the operation',
     type: { // THE PROBLEM PLACE
       count: {
         type: 'number',
         description: 'The number of instances updated',
       },
     },
     root: true,
   },
   http: {verb: 'post', path: '/update'},
 });
```

# Reproduction

To reproduce the problem, just run this test in any loopback application. It fails on `updateAll` shared method.

```
describe('Type Registry Tests', () => {
  let Model;

  before('init a memory datasource', () => {
    if (!app.dataSources.db) {
      app.dataSource('db', {connector: 'memory'});
    }
  });

  beforeEach('create a model and attach it to the app', () => {
    Model = app.registry.createModel({
      name: specs.modelName()
    });
    app.model(Model, {dataSource: 'db'});
  });

  it('should be compatible with get converter method', () => {
    _.forEach(app.models[Model.modelName].sharedClass.methods(), (method) => {
      should(() => {
        _.forEach(method.returns, ({type}) => {
          
          // The type is {count: {type: 'number', description: '...'} 
          // for updateAll method, but it supports only string and array
          app.remotes()._typeRegistry.getConverter(type);
        });
      }).not.throw();
    });
  });
});
```

# Additional information

"loopback": "3.17.0",
"strong-remoting": "3.6.0",

**Comment by jannyHou** [comment]
Hi @angfal I am trying to reproduce your error, are you running any endpoint from the Explorer(usually http://localhost:3000/explorer/)? If so can you post the endpoint name here? Thanks.

> I have a model with a remote datasource. When I tries to execute updateAll method my server crashes.

I am trying to figure out how do you execute the `updateAll` method.

**Comment by angfal** [comment]
@jannyHou I have two loopback servers A and B. The server A contains a model with a remote connector to the server B. When a function from the server A executes 'Model.updateAll' request to the server B, I receive success result, but the server A crashes when it tries to process the response.

I can provide a sandbox example if it doesn't help

**Comment by angfal** [comment]
The links to sandbox:
- https://github.com/angfal/remote-updateall-crash
or
- git@github.com:angfal/remote-updateall-crash.git

Just run the both servers and execute `Message.updateAll` service from the explorer of serverA (3000 port). The server crashes

**Comment by jannyHou** [comment]
@angfal Thank you now I can see the error, the `count`'s type is `number`, so its type is `string`, which should pass the following assertion:
```js
assert(typeof type === 'string' || Array.isArray(type),
    'type must be either an array or a string.');
```
Probably for some reason the type for `count` is detected incorrectly...looking into the bug. Will update here when have the fix PR.

**Comment by jannyHou** [comment]
FYI, I created a PR for it, but would like to hear opinions from `strong-remoting` experts before review.
https://github.com/strongloop/loopback/pull/3741

**Comment by angfal** [comment]
@jannyHou Thank you very much! But I made the same pull above. It was rejected

**Comment by jannyHou** [comment]
I see...closed https://github.com/strongloop/loopback/pull/3741, I find the rejected PR, following the chat there.

**Comment by angfal** [comment]
@jannyHou Any news here?

**Comment by ewrayjohnson** [comment]
Wanting to keep this open as I see no fix yet.

**Comment by bajtos** [comment]
Cross-posting https://github.com/strongloop/loopback/pull/3718#issuecomment-349319801:

> Hello,  thank you for the pull request. I am afraid we cannot revert #2842 because it's required for client code-generators to work, see the discussion in https://github.com/strongloop/loopback/issues/2104.
> 
> We need to fix strong-remoting and/or loopback-connector-remote to correctly support anonymously defined types in remote invocations.

Let me add `major` label to this issue to prevent stalebot from trying to automatically close it.

**Comment by agnes512** [comment]
Discussion from the estimation meeting:
1. check partition key settings
2. investigate how to pass partition key to drivers

**Comment by bajtos** [comment]
The fix: https://github.com/strongloop/strong-remoting/pull/472

**Comment by bajtos** [comment]
Fixed in `strong-remoting@3.15.0`
